src/app/store/slices/plan/index.ts

```diff
@@ -93,53 +82,69 @@ export const fetchUsageThunk = createAsyncThunk<UsageResponse | null, void, { st
     const isAuthenticated = getState().user.isAuthenticated;
 
     if (isAuthenticated) {
-      return await usageService.fetchUsage();
+      return usageService.fetchUsage();
     } else {
       return null;
     }
   },
 );
 
-export const fetchSubscriptionThunk = createAsyncThunk<UserSubscription | null, void, { state: RootState }>(
-  'plan/fetchSubscription',
-  async (payload: void, { getState }) => {
-    const isAuthenticated = getState().user.isAuthenticated;
+export const fetchSubscriptionThunk = createAsyncThunk<
+  UserSubscription | null,
+  { userType: UserType },
+  { state: RootState }
+>('plan/fetchSubscription', async (payload, { getState }) => {
+  const isAuthenticated = getState().user.isAuthenticated;
 
-    if (isAuthenticated) {
-      return paymentService.getUserSubscription();
-    } else {
-      return null;
+  if (isAuthenticated) {
+    return paymentService.getUserSubscription(payload.userType);
+  } else {
+    return null;
+  }
+});
+
+export const fetchBusinessLimitUsageThunk = createAsyncThunk<
+  GetMemberDetailsResponse | null,
+  void,
+  { state: RootState }
+>('plan/fetchBusinessLimitUsage', async (payload: void, { getState }) => {
+  const isAuthenticated = getState().user.isAuthenticated;
+  const userUuid = getState().user.user?.uuid;
+
+  if (isAuthenticated) {
+    const { selectedWorkspace } = getState().workspaces;
+
+    if (selectedWorkspace) {
+      const workspaceId = selectedWorkspace?.workspace.id;
+      if (workspaceId && userUuid) return workspacesService.getMemberDetails(workspaceId, userUuid);
```

Comment:
Here you seem to be returning a promise, handle this properly to return the required value. If you add the await I think that should be enough.